### PR TITLE
Fix test_grpc breakage from #290

### PR DESCRIPTION
**Pull request by glennmatthews**
- Remove argument validation test cases (now covered by spec tests)
- Fix Client.new calls to use named args.

```
TestGRPC#test_get_cli_cache = 3.14 s = .
TestGRPC#test_set_cli_string = 2.48 s = .
TestGRPC#test_smart_create = 1.15 s = .
TestGRPC#test_get_cli_incomplete = 1.05 s = .
E, [2016-03-15T13:28:21.276525 #16084] ERROR -- : initial connect failed: Failed authentication
TestGRPC#test_auth_failure = 0.85 s = .
Connection reset by peer? Try again
TestGRPC#test_set_cli_array = 3.58 s = .
TestGRPC#test_set_cli_invalid = 1.98 s = .
W, [2016-03-15T13:28:27.742290 #16084]  WARN -- : gRPC error '14' during 'show_cmd_text_output' request: 
W, [2016-03-15T13:28:27.742397 #16084]  WARN -- :   with cli: 'show clock'
W, [2016-03-15T13:28:27.742438 #16084]  WARN -- :   ''
E, [2016-03-15T13:28:27.742515 #16084] ERROR -- : initial connect failed: Connection refused: 
E0315 13:28:27.743896824   16084 tcp_client_posix.c:171]     failed to connect to 'ipv4:192.168.122.222:0': socket error: connection refused
E0315 13:28:28.745248875   16084 tcp_client_posix.c:171]     failed to connect to 'ipv4:192.168.122.222:0': socket error: connection refused
E0315 13:28:30.338694579   16084 tcp_client_posix.c:171]     failed to connect to 'ipv4:192.168.122.222:0': socket error: connection refused
E0315 13:28:32.526665582   16084 tcp_client_posix.c:171]     failed to connect to 'ipv4:192.168.122.222:0': socket error: connection refused
E0315 13:28:36.296423384   16084 tcp_client_posix.c:171]     failed to connect to 'ipv4:192.168.122.222:0': socket error: connection refused
W, [2016-03-15T13:28:37.744002 #16084]  WARN -- : gRPC error '4' during 'show_cmd_text_output' request: 
W, [2016-03-15T13:28:37.744179 #16084]  WARN -- :   with cli: 'show clock'
W, [2016-03-15T13:28:37.744223 #16084]  WARN -- :   'Deadline Exceeded'
E, [2016-03-15T13:28:37.744306 #16084] ERROR -- : initial connect failed: Deadline Exceeded
TestGRPC#test_connection_failure = 10.91 s = .
TestGRPC#test_get_cli_default = 1.17 s = .
TestGRPC#test_get_cli_invalid = 1.06 s = .
TestGRPC#test_get_cli_empty = 1.15 s = .
TestGRPC#test_get_cli_explicit = 0.96 s = .

Finished in 29.481501s, 0.4070 runs/s, 0.8480 assertions/s.

12 runs, 25 assertions, 0 failures, 0 errors, 0 skips
```

### CODE REVIEW

File: tests/test_grpc.rb [hunk 1]
```diff
@@ -85,14 +54,16 @@ def test_auth_failure
 
   def test_connection_failure
     # Connecting to a port that's listening, but not gRPC is one failure path
```
Comment (robert-w-gries) [inline tests/test_grpc.rb]: Nit: reword this comment to

``` diff
-     # Connecting to a port that's listening, but not gRPC is one failure path
+     # Connecting to a non-gRPC listening port is one failure path
```

Comment (glennmatthews) [inline tests/test_grpc.rb]: How about instead:

``` diff
-     # Connecting to a port that's listening, but not gRPC is one failure path
+     # Connecting to a port that's listening for a non-gRPC protocol is one failure path
```

Comment (robert-w-gries) [inline tests/test_grpc.rb]: Works for me

**Comment by robert-w-gries** [comment]
:+1:

File: tests/test_grpc.rb (hunk 1 above)
Comment (glennmatthews) [inline tests/test_grpc.rb]: Fixed.

**Pull request merged**